specs/NonClientHitTestKind.md

```diff
@@ -0,0 +1,182 @@
+Hit Test Kind
+===
+
+# Background
+The ability to ask the Webview for information on a hit test is an important feature for 
+visually hosted apps. It can be used to support features related to mouse activity in 
+regions that are traditionally Non-Client areas. Features like Draggable Regions, 
+Resize, Minimize/Maximize functionality, and more.
+
+
+# Examples
+## Draggable regions (HWND)
+Draggable regions are marked regions on a webpage that will move the app window when the 
+user clicks and drags on them. It also opens up the system menu if right-clicked. This 
+API is designed to provide draggable regions support for apps that are directly hosting 
+the CoreWebView2 via visual hosting and for UI framework WebView2 controls that use visual 
+hosting, like the WinUI2 and WinUI3 WebView2 control.If your app uses windowed hosting, 
+draggable regions are supported by default, no need to follow this guide.
+
+NOTE: Your app may already be forwarding mouse messages to the WebView, if so the bulk 
+of the code below is likely already in your app. One thing you need to do specifically 
+for this feature is to add the check for w_param == HTCAPTION as a clause for capturing 
+the mouse and forwarding messages. It is necessary for the system menu to show. If you 
+choose not to include this you would have to handle the logic for showing the system menu 
+yourself.
+
+NOTE: This API is not currently supported for environments that use a CoreWindow like UWP
+
+#### Win32 C++
+```cpp
+LRESULT CALLBACK WndProc(HWND hwnd, UINT message, WPARAM wParam, LPARAM lParam)
+{
+....
+// The following code forwards mouse messages to the WebView2 - Start
+
+// used to control when messages are forwarded to WebView
+bool isCapturingMouse = false; 
+
+// handle the range of mouse messages
+if (message >= WM_MOUSEFIRST && message <= WM_MOUSELAST) 
+{
+    POINT point {GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam)};
+    // IsPointInWebview is an example of a function which your app can implement that checks if 
+    // the mouse message point lies inside the rect of the visual thosts the WebView
+    if (IsPointInWebView(point) || message == WM_MOUSELEAVE || isCapturingMouse) 
```

Comment:
isCapturingMouse is set false above and so will always be false here in this if check. Is this an error?

Reply:
I don't believe so, all clauses of the if are disjunctive so even when isCapturingMouse is false, IsPointInWebview can return true and the if statement will be true.  This piece of code is typical for visually hosted apps that wish to forward mouse messages to WV. The logic was taken from our win32 sample app, in ViewComponent.cpp

Reply:
In the ViewComponent, isCaptureMouse is a member variable (m_isCapturingMouse) so that the state of it persists across calls to OnMouseMessage. In this sample code, isCaptureMouse is a local variable that is always set to false at the start of the function.